txstratum/healthcheck.py

```diff
@@ -0,0 +1,265 @@
+
+    It will:
+    - Check that the manager has at least 1 miner
+    - Check that all 'tx_jobs' in the manager in the last 5 minutes have a status of done
+      and have a total_time lesser than 10 seconds.
+    - Check that all miners have submitted at least 1 job in the last 1 hour.
+      Note that subimitting a job doesn't mean solving a tx.
+
+    If at least one of the 'tx_jobs' has a 'total_time' of more than 10 seconds, the health check
+    will be returned as 'warn'.
+
+    If some of the 'tx_jobs' has status different than 'done', the health check will be returned
+    as 'fail'.
+
+    If there are no miners, the health check will be returned as 'fail'
+
+    If there are miners, but any of them has submitted a job in the last 1 hour, the health check
+    will be returned as 'fail'
+    """
+
+    def __init__(self, manager: "TxMiningManager") -> None:
+        """Init the class with the manager instance."""
+        self.manager = manager
+        self.last_manager_status = ComponentHealthCheck(
+            component_name="manager",
+            component_type=ComponentType.INTERNAL,
+            status=HealthCheckStatus.PASS,
+            output="everything is ok",
+        )
+
+    async def get_health_check(self) -> ComponentHealthCheck:
+        """Return the manager health check status."""
+        # Check that the manager has at least 1 miner
+        if not self.manager.has_any_miner():
+            return ComponentHealthCheck(
+                component_name="manager",
+                component_type=ComponentType.INTERNAL,
+                status=HealthCheckStatus.FAIL,
+                output="no miners connected",
+            )
+
+        # Check that all the miners submitted in the last 1 hour. If not, return failed.
+        if not self.manager.has_any_submitted_job_in_period(period=3600):
+            return ComponentHealthCheck(
+                component_name="manager",
+                component_type=ComponentType.INTERNAL,
+                status=HealthCheckStatus.FAIL,
+                output="no miners submitted a job in the last 1 hour",
+            )
+
+        if not self.manager.tx_jobs:
+            # We just return the last saved status in case there are no jobs in the last 5 minutes
+            return self.last_manager_status
+
+        for job in self.manager.tx_jobs.values():
+            if job.is_failed():
+                self.last_manager_status.update(
+                    {
+                        "status": HealthCheckStatus.FAIL,
+                        "output": "some tx_jobs in the last 5 minutes have failed",
+                    }
+                )
+
+                return self.last_manager_status
+            if job.is_done():
+                assert (
+                    job.total_time is not None
+                ), "total_time must be set if job is done"
+
+                if job.total_time > 10:
```

Comment:
Move 10 to a constant?

Reply:
Done in https://github.com/HathorNetwork/tx-mining-service/pull/104/commits/acc340e690cfd4bf508fa18b6b25a2a98385b5aa